src/java/org/apache/cassandra/db/RowIndexEntry.java

```diff
@@ -352,31 +353,31 @@ public RowIndexEntry<IndexInfo> deserialize(DataInputPlus in, long indexFilePosi
         private void checkSize(int entries, int bytes)
         {
             ReadCommand command = ReadCommand.getCommand();
-            if (command == null || SchemaConstants.isSystemKeyspace(command.metadata().keyspace) || !DatabaseDescriptor.getTrackWarningsEnabled())
+            if (command == null || SchemaConstants.isSystemKeyspace(command.metadata().keyspace) || !DatabaseDescriptor.getReadThresholdsEnabled())
```

Comment:
`getReadThresholdsEnabled` should be faster than `isSystemKeyspace`, we could change the order to check `getReadThresholdsEnabled` first.

Reply:
I rather leave as is for this patch.

> getReadThresholdsEnabled should be faster than isSystemKeyspace

🤷 may or may not, 3 set contains clauses on non-thread safe data structures vs a memory fence... without measuring I could see it go both ways